src/ngraph/frontend/onnxifi/backend.hpp

```diff
@@ -0,0 +1,86 @@
+//*****************************************************************************
+// Copyright 2017-2018 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//*****************************************************************************
+
+#pragma once
+
+#include <memory>  // std::shared_ptr
+#include <string>  // std::string
+#include <utility> // std::move
+#include <vector>  // std::vector
+
+#include "ngraph/function.hpp"
+#include "ngraph/runtime/backend.hpp"
+#include "ngraph/runtime/tensor_view.hpp"
+
+namespace ngraph
+{
+    namespace onnxifi
+    {
+        /// \brief ONNXIFI extensions to nGraph backend
+        class Backend
```

Comment:
How one would get id for _specific_ `Backend`?  There are some ONNXIFI functions that require backend id i.e. `onnxGetBackendInfo` or  `onnxInitBackend`

Reply:
Please use `onnxGetBackendIDs()` to obtain IDs and `onnxGetBackendInfo()` to choose the backend (based on properties you choose).

Reply:
or to use `onnxGetBackendCompatibility()` instead of `onnxGetBackendInfo()`